src/test/UniswapV3SplitRouteCallee.t.sol

```diff
@@ -0,0 +1,375 @@
+        }
+
+        // Vyper-style
+        for (uint256 i = 0; i < 20; i++) {
+            // Scan the storage for the balance storage slot
+            bytes32 prevValue = hevm.load(token, keccak256(abi.encode(uint256(i), address(this))));
+            hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), bytes32(amount));
+            if (GemAbstract(token).balanceOf(address(this)) == amount) {
+                // Found it
+                return;
+            } else {
+                // Keep going after restoring the original value
+                hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), prevValue);
+            }
+        }
+    }
+
+    address wethAddr;
+    address linkAddr;
+    address daiAddr;
+    address vatAddr;
+    address linkJoinAddr;
+    address daiJoinAddr;
+    address dogAddr;
+    address jugAddr;
+    address linkClipAddr;
+    address linkPipAddr;
+
+    Hevm hevm;
+    WethAbstract    weth;
+    GemAbstract     link;
+    VatAbstract     vat;
+    DaiAbstract     dai;
+    GemJoinAbstract linkJoin;
+    DogAbstract     dog;
+    JugAbstract     jug;
+    ClipAbstract    linkClip;
+
+    OsmAbstract linkPip;
+
+    function setAddresses() private {
+        ChainlogHelper helper = new ChainlogHelper();
+        ChainlogAbstract chainLog = helper.ABSTRACT();
+        wethAddr = chainLog.getAddress("ETH");
+        linkAddr = chainLog.getAddress("LINK");
+        vatAddr = chainLog.getAddress("MCD_VAT");
+        daiAddr = chainLog.getAddress("MCD_DAI");
+        linkJoinAddr = chainLog.getAddress("MCD_JOIN_LINK_A");
+        daiJoinAddr = chainLog.getAddress("MCD_JOIN_DAI");
+        dogAddr = chainLog.getAddress("MCD_DOG");
+        jugAddr = chainLog.getAddress("MCD_JUG");
+        linkClipAddr = chainLog.getAddress("MCD_CLIP_LINK_A");
+        linkPipAddr = chainLog.getAddress("PIP_LINK");
+    }
+
+    function setInterfaces() private {
+        hevm = Hevm(hevmAddr);
+        weth = WethAbstract(wethAddr);
+        link = GemAbstract(linkAddr);
+        vat = VatAbstract(vatAddr);
+        dai = DaiAbstract(daiAddr);
+        linkJoin = GemJoinAbstract(linkJoinAddr);
+        dog = DogAbstract(dogAddr);
+        jug = JugAbstract(jugAddr);
+        linkClip = ClipAbstract(linkClipAddr);
+        linkPip = OsmAbstract(linkPipAddr);
+    }
+
+    VaultHolder          ali;
+    UniswapV3SplitCallee dan;
+
+    address aliAddr;
+    address danAddr;
+    address UniswapV3Router2;
+    bytes   uniswapTxDataProfit;
+    bytes   uniswapTxDataNoProfit;
+
+    function getPermissions() private {
+        hevm.store(dogAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(vatAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(linkPipAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        linkPip.kiss(address(this));
+    }
+
+    function setEnvVars() private {
```

Comment:
Please add an explanation as in 1inch test.

Reply:
added